crt/fork.c

```diff
@@ -0,0 +1,427 @@
+#ifdef MYST_PSEUDO_FORK
+#define hidden __attribute__((__visibility__("hidden")))
+
+#define _GNU_SOURCE
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <linux/futex.h>
+#include <pthread.h>
+#include <sched.h>
+#include <signal.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <pthread_impl.h>
+
+#include <myst/setjmp.h>
+#include <myst/syscallext.h>
+
+static int _round_up(uint64_t x, uint64_t m, uint64_t* r)
+{
+    uint64_t t;
+
+    if (!r)
+        return -EINVAL;
+
+    /* prevent divide by zero */
+    if (m == 0)
+        return -ERANGE;
+
+    if (__builtin_add_overflow(x, m - 1, &t))
+        return -ERANGE;
+
+    if (__builtin_mul_overflow(t / m, m, r))
+        return -ERANGE;
+
+    return 0;
+}
+
+/* ATTN: does not work for Linux yet: SYS_arch_prctl */
+static void _set_fsbase(void* p)
+{
+    __asm__ volatile("wrfsbase %0" ::"r"(p));
+}
+
+extern hidden unsigned __default_guardsize;
+extern hidden unsigned __default_stacksize;
+extern hidden volatile size_t __pthread_tsd_size;
```

Comment:
These three symbols are declared in ``./musl/src/internal/pthread_impl.h``. Is there some way to include it?

Reply:
We do actually include that header now. I'll delete